test/unit/cluster_resource_discovery_test.rb

```diff
@@ -2,41 +2,62 @@
 require 'test_helper'
 
 class ClusterResourceDiscoveryTest < Krane::TestCase
-  def test_global_resource_kinds_failure
+  def test_fetch_resources_failure
     crd = mocked_cluster_resource_discovery(nil, success: false)
-    kinds = crd.global_resource_kinds
-    assert_equal(kinds, [])
+    resources = crd.fetch_resources
+    assert_equal(resources, [])
   end
 
-  def test_global_resource_kinds_success
-    crd = mocked_cluster_resource_discovery(api_resources_full_response)
-    kinds = crd.global_resource_kinds
-    assert_equal(kinds.length, api_resources_full_response.split("\n").length - 1)
+  def test_fetch_resources_not_namespaced
+    crd = mocked_cluster_resource_discovery(api_resources_not_namespaced_full_response)
+    kinds = crd.fetch_resources(namespaced: false).map { |r| r['kind'] }
+    assert_equal(kinds.length, api_resources_not_namespaced_full_response.split("\n").length - 1)
     %w(MutatingWebhookConfiguration ComponentStatus CustomResourceDefinition).each do |kind|
       assert_includes(kinds, kind)
     end
   end
 
-  def test_prunable_resources
+  def test_fetch_resources_namespaced
+    crd = mocked_cluster_resource_discovery(api_resources_namespaced_full_response, namespaced: true)
+    kinds = crd.fetch_resources(namespaced: true).map { |r| r['kind'] }
+    assert_equal(kinds.length, api_resources_namespaced_full_response.split("\n").length - 1)
+    %w(ConfigMap CronJob Deployment).each do |kind|
+      assert_includes(kinds, kind)
+    end
+  end
+
+  def test_prunable_global_resources
     Krane::Kubectl.any_instance.stubs(:run).with("api-versions", attempts: 5, use_namespace: false)
       .returns([api_versions_full_response, "", stub(success?: true)])
-    crd = mocked_cluster_resource_discovery(api_resources_full_response)
+    crd = mocked_cluster_resource_discovery(api_resources_not_namespaced_full_response)
     kinds = crd.prunable_resources(namespaced: false)
 
     assert_equal(kinds.length, 13)
-    %w(scheduling.k8s.io/v1beta1/PriorityClass storage.k8s.io/v1beta1/StorageClass).each do |kind|
-      assert_includes(kinds, kind)
+    %w(PriorityClass StorageClass).each do |expected_kind|
+      assert kinds.one? { |k| k.include?(expected_kind) }
```

Comment:
Is this just a sanity check or is there a reason for these 2 in particular?

Reply:
Just a sanity check, the `assert_equal(kinds.length, 13)` is the check to ensure list has the right number of items.